pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/UpdateActivityContentSourceDB.java

```diff
@@ -0,0 +1,696 @@
+
+        //load i18n
+        //todo pass file path as params
+        //need controls to block running on studies accidentally
+        i18nCfgEn = ConfigFactory.parseFile(Paths.get(enFilePath).toFile());
+        i18nCfgEs = ConfigFactory.parseFile(Paths.get(esFilePath).toFile());
+
+        templateUpdateList = new HashSet<>();
+
+        //template with multiple substitutions
+        //should we group ?
+        //load dataCfg
+        List<? extends Config> configList = dataCfg.getConfigList(TEMPLATE_UPDATES);
+        for (Config tmplUpdateCfg : configList) {
+            TemplateUpdateInfo tmplUpdInfo = gson.fromJson(ConfigUtil.toJson(tmplUpdateCfg), TemplateUpdateInfo.class);
+            templateUpdateList.add(tmplUpdInfo);
+        }
+
+        //Maps to save ALL translations for any verification
+        allActTransMapEN = new TreeMap<String, Map>();
+        allActTransMapES = new TreeMap<String, Map>();
+        interestedTransVarsMapEN = new TreeMap<String, Set<TemplateVariable>>();
+        interestedTemplatesMapEN = new TreeMap<String, Set<Template>>();
+        //var activityBuilder = new ActivityBuilder(cfgPath.getParent(), studyCfg, varsCfg, studyDto, adminUser.getId());
+
+        //traverseEventConfigurations(handle, studyDto.getId());
+        traverseActivities(handle, studyDto);
+        String gsonDataTemplatesEN = gson.toJson(interestedTemplatesMapEN, typeObject);
+        String gsonDataVarsEN = gson.toJson(interestedTransVarsMapEN, typeObject);
+        log.info("Reverse engineered EN Matching templates");
+        log.info(gsonDataTemplatesEN);
+        log.info("Reverse engineered EN Matching template VARS");
+        log.info(gsonDataVarsEN);
+
+        //String gsonDataEN = gson.toJson(allActTransMapEN, typeObject);
+        //log.info("Reverse engineered EN translation file");
+        //log.info(gsonDataEN);
+
+        //log.info("Reverse engineered ES translation file");
+        //String gsonDataES = gson.toJson(allActTransMapES, typeObject);
+        //log.info(gsonDataES);
+
+    }
+
+    private void revisionTemplate(Template template, RevisionMetadata meta, ActivityVersionDto newVersionDto,
+                                  Set<TemplateUpdateInfo> updateInfos) {
+
+        //for now considering as body_template and not title_template
+        //todo .. query DB and act accordingly
+
+        //steps: revision the block, revision the template, insert new template, create new block_content
+
+        //iterate through and apply ALL changes
+        String newTemplateText = template.getTemplateText();
+        log.info("revisioning and updating template : {}", template.getTemplateText());
+        for (TemplateUpdateInfo updateInfo : updateInfos) {
+            newTemplateText = newTemplateText.replace(updateInfo.getSearchString(), updateInfo.getReplaceString());
+            newTemplateText = newTemplateText.replace("<li></li>", ""); //remove empty tags if any
```

Comment:
Is there a way to generalize this with jsoup so we can remove any empty element?

Reply:
That was result of removing one PI name . 
Actually I could just add these tags to the search strings and that eliminates this need. 
updated the patch file and removed this line